tests/helper/prism-loader.js

```diff
@@ -26,20 +52,81 @@ module.exports = {
 		return context.Prism;
 	},
 
+	/**
+	 * Creates a new JavaScript DOM instance with Prism being loaded.
+	 *
+	 * @returns {PrismDOM}
+	 */
+	createPrismDOM() {
+		const dom = new JSDOM(``, {
+			runScripts: 'outside-only'
+		});
+		const window = dom.window;
+
+		const loadLanguageSource = this.loadComponentSource.bind(this);
+		const loadPluginSource = this.loadPluginSource.bind(this);
+
+		window.self = window; // set self for plugins
+		window.eval(loadLanguageSource('core'));
+
+		const loadedLanguages = new Set();
+		const loadedPlugins = new Set();
+
+		return {
+			window: /** @type {PrismWindow} */ (window),
+
+			loadLanguages: function loadLanguages(languages) {
+				for (const language of toArray(languages)) {
+					if (!(language in languagesCatalog)) {
+						throw new Error(`Unknown language '${language}'`);
+					}
+
+					if (loadedLanguages.has(language)) {
+						continue;
+					}
+					loadedLanguages.add(language);
+
+					const require = languagesCatalog[language].require;
+					if (require) {
+						loadLanguages(require);
+					}
+
+					window.eval(loadLanguageSource(language));
```

Comment:
This function can also be used with the dependency loader,  ya? We should really get that in.